www/template.html

```diff
@@ -25,22 +25,17 @@
           instance.exports.wasmInit();
 
           let frame = true;
+          let last_update_time = performance.now();
           let update = function() {{
             if (!frame) return;
-            instance.exports.wasmUpdate();
-            if (mach.wait_event_timeout > 0) {{
-              mach.wait_event_timer = setTimeout(() => {{
-                window.requestAnimationFrame(update);
-                }},
-                mach.wait_event_timeout * 1000);
-            }}
-            else {{
-              window.requestAnimationFrame(update);
+            if (mach.machHasEvent() ||
```

Comment:
This is perfect, but with one problem is this will always wait, as in we current can't just poll without waiting. This needs the same type of 3 way comparision as native:
```zig
if (mach.wait && (mach.machHasEvent() || (...)) {

} else {
    rAF(); // Current case, default, poll but don't wait
}
```
The ``mach.wait`` field will be set by Engine.setWaitTimeout();

Reply:
No wait @david-vanderson your'e right. Its in OR.